arm-appinsights/2015-05-01/swagger/component_API.json

```diff
@@ -0,0 +1,376 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when creating or updating an Application Insights component. The updated component is returned.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponent"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentCreate": { "$ref": "../examples/ComponentCreate.json" },
+          "ComponentUpdate": { "$ref": "../examples/ComponentUpdate.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "ApplicationInsightsComponent": {
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The kind of application that this component refers to, used to customize UI. This value is a freeform string, values should typically be one of the following: web, ios, other, store, java, phone."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationInsightsComponentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "kind"
+      ],
+      "description": "An Application Insights component definition."
+    },
+    "ApplicationInsightsComponentProperties": {
+      "description": "Properties that define an Application Insights component resource.",
+      "properties": {
+        "ApplicationId": {
+          "type": "string",
+          "description": "The unique ID of your application."
+        },
+        "AppId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Application Insights Unique ID for your Application."
+        },
+        "Application_Type": {
+          "type": "string",
+          "description": "Type of application being monitored. Example values: Web, other. Cannot be changed after the component has been created."
+        },
+        "Flow_Type": {
```

Comment:
This is very helpful, and is useful when creating new Components, they do indeed all get created with the 'Bluefield' value. However, when I get a list of components, I get components that have different FlowTypes (ie. 'Redfield'). In the C# SDK that I generate from my swagger, I now get deserialization exceptions.

Apologies for the confusion, perhaps this will help?

My question then changes to: 
Can I specify in my swagger spec a way to ensure that FlowType is always *set* to 'Bluefield' but that still allows the *read* of other values of FlowType. It is always acceptable for us to set the field to 'Bluefield' if we ever update a component that had a different value previously.